src/test/java/jenkins/plugins/openstack/compute/JCloudsRetentionStrategyTest.java

```diff
@@ -129,4 +131,56 @@ public void doNotDeleteSlavePutOfflineByUser() throws Exception {
         computer.getRetentionStrategy().check(computer);
         assertTrue(computer.isPendingDelete());
     }
+
+    @Test
+    public void doNotDeleteNewSlaveIfInstanceRequired() throws Exception {
+        JCloudsCloud cloud = j.configureSlaveLaunching(j.dummyCloud(j.dummySlaveTemplate(
+            j.defaultSlaveOptions().getBuilder().retentionTime(0).instancesMin(1).build(),
+            "label"
+        )));
+        JCloudsSlave slave = j.provision(cloud, "label");
+        JCloudsComputer computer = (JCloudsComputer) slave.toComputer();
+        computer.waitUntilOnline();
+
+        computer.getRetentionStrategy().check(computer);
+
+        assertFalse(computer.isPendingDelete());
+    }
+
+    @Test
+    public void deleteUsedSlaveWhenOnlyNewInstancesAreRequired() throws Exception {
+        JCloudsCloud cloud = j.configureSlaveLaunching(j.dummyCloud(j.dummySlaveTemplate(
+            j.defaultSlaveOptions().getBuilder().retentionTime(0).instancesMin(1).build(),
+            "label"
+        )));
+        JCloudsSlave slave = j.provision(cloud, "label");
+        JCloudsComputer computer = (JCloudsComputer) slave.toComputer();
+        computer.waitUntilOnline();
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.setAssignedNode(slave);
+        FreeStyleBuild build = p.scheduleBuild2(0).waitForStart();
+        j.waitForCompletion(build);
+
+        computer.getRetentionStrategy().check(computer);
+
+        assertTrue(computer.isPendingDelete());
+    }
+
+    @Test
```

Comment:
Please, fix the indentation.

Reply:
I suppose you're referring to the bit in `JCloudsCloud cloud = j.configureSlaveProvisioning(j.dummyCloud(j.dummySlaveTemplate(`: done.